fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteSimpleAggToHDFSScanRule.java

```diff
@@ -0,0 +1,191 @@
+            }
+        }
+
+        ColumnRefOperator placeholderColumn = null;
+
+        for (Map.Entry<ColumnRefOperator, CallOperator> kv : aggs.entrySet()) {
+            CallOperator aggCall = kv.getValue();
+            // ___count___
+            String metaColumnName = "___" + aggCall.getFnName() + "___";
+            Type columnType = aggCall.getType();
+
+            if (placeholderColumn == null) {
+                placeholderColumn = columnRefFactory.create(metaColumnName, columnType, aggCall.isNullable());
+                Column c = new Column();
+                c.setName(metaColumnName);
+                c.setIsAllowNull(true);
+                newScanColumnRefs.put(placeholderColumn, c);
+            }
+
+            Function aggFunction = aggCall.getFunction();
+            String newAggFnName = aggCall.getFnName();
+            Type newAggReturnType = aggCall.getType();
+            if (aggCall.getFnName().equals(FunctionSet.COUNT)) {
+                aggFunction = Expr.getBuiltinFunction(FunctionSet.SUM,
+                        new Type[] {Type.BIGINT}, Function.CompareMode.IS_IDENTICAL);
+                newAggFnName = FunctionSet.SUM;
+                newAggReturnType = Type.BIGINT;
+            }
+            CallOperator newAggCall = new CallOperator(newAggFnName, newAggReturnType,
+                    Collections.singletonList(placeholderColumn), aggFunction);
+            newAggCalls.put(kv.getKey(), newAggCall);
+        }
+
+        Map<Column, ColumnRefOperator> newScanColumnMeta = Maps.newHashMap();
+        for (Map.Entry<ColumnRefOperator, Column> c : newScanColumnRefs.entrySet()) {
+            newScanColumnMeta.put(c.getValue(), c.getKey());
+        }
+
+        LogicalScanOperator newMetaScan = null;
+
+        if (scanOperator instanceof LogicalHiveScanOperator) {
+            newMetaScan = new LogicalHiveScanOperator(scanOperator.getTable(),
+                    newScanColumnRefs, newScanColumnMeta, scanOperator.getLimit(), scanOperator.getPredicate());
+        } else if (scanOperator instanceof LogicalIcebergScanOperator) {
+            newMetaScan = new LogicalIcebergScanOperator(scanOperator.getTable(),
+                    newScanColumnRefs, newScanColumnMeta, scanOperator.getLimit(), scanOperator.getPredicate());
+        } else {
+            throw new IllegalStateException("Unknown scan operator: " + scanOperator);
+        }
+        try {
+            newMetaScan.setScanOperatorPredicates(scanOperator.getScanOperatorPredicates());
+        } catch (AnalysisException e) {
+            throw new IllegalStateException("Set scan operator predicates", e);
+        }
+        LogicalAggregationOperator newAggOperator = new LogicalAggregationOperator(aggregationOperator.getType(),
+                aggregationOperator.getGroupingKeys(), newAggCalls);
+
+        newAggOperator.setProjection(aggregationOperator.getProjection());
```

Comment:
why set projection? I think there is no projection, just logical project node above agg node

Reply:
I'm not sure here. I guess there is no harm to do that.